### ISSUE TITLE
The ability to extend the declarative style of the subviews is very limited

### ISSUE THREAD

**Issue by dschissler** [AmpersandJS/ampersand-view#150]
https://github.com/AmpersandJS/ampersand-view/blob/master/ampersand-view.js#L272-L280

I'm currently overloading this entire function but all that I really want to overload is the `var opts ...` part.  That part is the essence of the declarative format and the `action` function makes sure that it happens in the correct way and I don't want to change that.  So it seems that we must use a very basic declarative format with a `prepareView` function.  I think that great things are possible if this is broken out with a few lines of comments describing how it can be overloaded to extend the declarative subview format.

What does everyone think of this?  Then one would just overload this method without having to take all of the inline `action` function junk with it.

```javascript
    _parseSubviewOpts: function (subview) {
        var opts = {
            selector: subview.selector || '[data-hook="' + subview.hook + '"]',
            waitFor: subview.waitFor || '',
            prepareView: subview.prepareView || function () {
                return new subview.constructor({
                    parent: self
                });
            }
        };
        return opts;
    }
```

**Comment by cdaringe** [comment]
I think that's just fine :smile:

**Comment by dschissler** [comment]
I created a [pull request](https://github.com/AmpersandJS/ampersand-view/pull/151) for this feature.  I'm using it now and it appears to work.

**Comment by dschissler** [comment]
Please get it in there and I'll write up more comprehensive documentation and I'll fix the obvious grammar mistake as well ;).

**Comment by dschissler** [comment]
I'm using my enhancement in the following ways:

In this case I use `waitForData` to make something like a simple link just render immediately.

```javascript
  subviews: {
    recordLink: {
      constructor: FileExternalLinkView,
      hook: 'record-link',
      waitForData: {
        model: 'file[0]'
      }
    }
  }
```

In this case I'm allowing a list view to render and then to have its collection rendered once the data is propagated through:

```javascript
  subviews: {
    list: {
      constructor: SelectableListView,
      hook: 'list',
      data: {
        collection: 'list'
      }
    }
  }
 ```

The idea is to give me flexibility and to prevent unnecessary DOM hits.  A simple link should be able to render in one go and what is the point if the data isn't ready?

**Comment by dschissler** [comment]
@cdaringe I removed the unnecessary parameter and expanded the documentation.  This is a very low impact change and I'd just like to get it in there and have it pushed to npm.  I have some loose plans on expanding this but I'll need to be much farther along in my project to get into that.  It will be in some months and there is nothing wrong with this change as a stand alone.

**Comment by cdaringe** [comment]
published as `9.0.1`.

thanks @dschissler!
